Add recipe for librobometry

traversaro opened the pull request:
<!--
Thank you very much for putting in this recipe PR!

This repository is very active, so if you need help with a PR, please let the
right people know. There are language-specific teams for reviewing recipes.

| Language        | Name of review team           |
| --------------- | ----------------------------- |
| python          | `@conda-forge/help-python`    |
| python/c hybrid | `@conda-forge/help-python-c`  |
| r               | `@conda-forge/help-r`         |
| java            | `@conda-forge/help-java`      |
| nodejs          | `@conda-forge/help-nodejs`    |
| c/c++           | `@conda-forge/help-c-cpp`     |
| perl            | `@conda-forge/help-perl`      |
| Julia           | `@conda-forge/help-julia`     |
| ruby            | `@conda-forge/help-ruby`      |
| other           | `@conda-forge/staged-recipes` |

Once the PR is ready for review, please mention one of the teams above in a
new comment. i.e. `@conda-forge/help-some-language, ready for review!`
Then, a bot will label the PR as 'review-requested'.

Due to GitHub limitations, first time contributors to conda-forge are unable
to ping conda-forge teams directly, but you can [ask a bot to ping the team][1]
using a special command in a comment on the PR to get the attention of the
`staged-recipes` team. You can also consider asking on our [Gitter channel][2]
if your recipe isn't reviewed promptly.

[1]: https://conda-forge.org/docs/maintainer/infrastructure.html#conda-forge-admin-please-ping-team
[2]: https://gitter.im/conda-forge/conda-forge.github.io
-->

Add a recipe for the [robometry C++ library](https://github.com/robotology/robometry). Given the discussion in https://github.com/conda-forge/staged-recipes/pull/19764#pullrequestreview-1085836936, the recipe and conda package is called `librobometry`.

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/librobometry```) and found it was in an excellent condition.

traversaro commented:
Due to the use of std::filesystem, on macOS the library needs to use 10.15 SDK, otherwise the following error appears:
~~~
[3/10] Building CXX object src/librobometry/CMakeFiles/robometry.dir/src/BufferManager.cpp.o
FAILED: src/librobometry/CMakeFiles/robometry.dir/src/BufferManager.cpp.o 
$BUILD_PREFIX/bin/x86_64-apple-darwin13.4.0-clang++ -DBOOST_ALL_NO_LIB -DROBOMETRY_EXPORTS -I$SRC_DIR/src/librobometry/include -isystem $PREFIX/include/eigen3 -march=core2 -mtune=haswell -mssse3 -ftree-vectorize -fPIC -fPIE -fstack-protector-strong -O2 -pipe -stdlib=libc++ -fvisibility-inlines-hidden -fmessage-length=0 -isystem $PREFIX/include -fdebug-prefix-map=$SRC_DIR=/usr/local/src/conda/librobometry-1.1.0 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix -O3 -DNDEBUG -isysroot /Applications/Xcode_13.2.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.13.sdk -mmacosx-version-min=10.13 -fPIC -std=c++17 -MD -MT src/librobometry/CMakeFiles/robometry.dir/src/BufferManager.cpp.o -MF src/librobometry/CMakeFiles/robometry.dir/src/BufferManager.cpp.o.d -o src/librobometry/CMakeFiles/robometry.dir/src/BufferManager.cpp.o -c $SRC_DIR/src/librobometry/src/BufferManager.cpp
$SRC_DIR/src/librobometry/src/BufferManager.cpp:88:61: error: 'exists' is unavailable: introduced in macOS 10.15
$SRC_DIR/src/librobometry/src/BufferManager.cpp:88:68: error: 'path' is unavailable: introduced in macOS 10.15
    if (!m_bufferConfig.path.empty() && !yarp_telemetry_fs::exists(m_bufferConfig.path)) {
                                                                   ^
$BUILD_PREFIX/bin/../include/c++/v1/__filesystem/path.h:436:3: note: 'path' has been explicitly marked unavailable here
  path(const _Source& __src, format = format::auto_format) {
  ^
$SRC_DIR/src/librobometry/src/BufferManager.cpp:88:68: error: '~path' is unavailable: introduced in macOS 10.15
    if (!m_bufferConfig.path.empty() && !yarp_telemetry_fs::exists(m_bufferConfig.path)) {
                                                                   ^
$BUILD_PREFIX/bin/../include/c++/v1/__filesystem/path.h:458:3: note: '~path' has been explicitly marked unavailable here
  ~path() = default;
  ^
$SRC_DIR/src/librobometry/src/BufferManager.cpp:90:47: error: 'create_directory' is unavailable: introduced in macOS 10.15
        auto dir_created = yarp_telemetry_fs::create_directory(m_bufferConfig.path, ec);
                                              ^
$BUILD_PREFIX/bin/../include/c++/v1/__filesystem/operations.h:80:39: note: 'create_directory' has been explicitly marked unavailable here
inline _LIBCPP_INLINE_VISIBILITY bool create_directory(const path& __p, error_code& __ec) noexcept { return __create_directory(__p, &__ec); }
                                      ^
$SRC_DIR/src/librobometry/src/BufferManager.cpp:90:64: error: 'path' is unavailable: introduced in macOS 10.15
        auto dir_created = yarp_telemetry_fs::create_directory(m_bufferConfig.path, ec);
                                                               ^
$BUILD_PREFIX/bin/../include/c++/v1/__filesystem/path.h:436:3: note: 'path' has been explicitly marked unavailable here
  path(const _Source& __src, format = format::auto_format) {
  ^
$SRC_DIR/src/librobometry/src/BufferManager.cpp:90:64: error: '~path' is unavailable: introduced in macOS 10.15
        auto dir_created = yarp_telemetry_fs::create_directory(m_bufferConfig.path, ec);
                                                               ^
$BUILD_PREFIX/bin/../include/c++/v1/__filesystem/path.h:458:3: note: '~path' has been explicitly marked unavailable here
  ~path() = default;
  ^
6 errors generated.
~~~

traversaro commented:
@conda-forge/help-c-cpp The PR is ready for review.
